Backend/AlgorithmsManager.py

```diff
@@ -513,36 +492,26 @@ def sort_tweets(arr):
 
             arr[k + 1] = temp
 
-
+    # pass in a dictionary and the length of the returned shortlist
+    # return a list of tuples, sorted by the retweet count
     @staticmethod
-    def sort_retweeters(arr):
-        length = len(arr)
-        for l in range(1, length):
-            temp = arr[l]
-            tempval = temp['retweets_count']
-            k = l-1
-
-            num_trend_sort_iter = 0
-
-            while k >= 0 and    (tempval if tempval is not None else 0) >= \
-                                (arr[k]['retweets_count'] if arr[k]['retweets_count'] is not None else 0):
-                arr[k+1] = arr[k]
-                k -= 1
-
-                # todo strip debugging code from while loop
-                num_trend_sort_iter +=1
-                if num_trend_sort_iter >= 10000:
-                    print("while loop started to go infinite:\nARGUMENTS:\n",
-                          "arr= ", arr,
-                          ";\n\nVARIABLES:\n",
-                          "temp= ", temp, "\ntempval= ", tempval, "\nk= ", k,
-                          "\nSTACK TRACE:\n")
-
-                    traceback.print_stack()
-                    raise Exception("got into infinite loop")
-
-            arr[k+1] = temp
-
+    def sort_retweeters(dict, count):
```

Comment:
As this function has changed, I feel like the name is no longer as clear. Would you consider renaming it to something like 'list_top_retweeters'?